day_trader/exchange/models.py

```diff
@@ -74,23 +109,72 @@ class BuyTrigger(models.Model):
 
     def check_validity(self, price, symbol):
         if(self.price >= price):
-            user_stock = UserStock.objects.get(
-                user_id=self.user_id, stock_symbol=symbol)
-            user_stock.update_amount(cash_amount//self.price)
+            user = User.objects.get(user_id=trigger.user_id)
+            buy = Buy.create(user_id=user.user_id, stock_symbol=self.stock_symbol, cash_amount=cash_amount, stock_price=price)
+            buy.commit(user)
             self.committed = True
             self.save()
 
 
 class Sell(models.Model):
     user_id = models.ForeignKey(User, on_delete=models.CASCADE)
     stock_symbol = models.CharField(max_length=3, primary_key=True)
-    cash_amount = models.DecimalField(
+    intended_cash_amount = models.DecimalField(
+        max_digits=65, decimal_places=2, default=0)
+    actual_cash_amount = models.DecimalField(
         max_digits=65, decimal_places=2, default=0)
     stock_sold_amount = models.PositiveIntegerField(default=0)
 
+    @classmethod
+    def create(cls, user_id, stock_symbol, cash_amount, stock_price, user):
+        sell = cls(user_id=user_id, stock_symbol=stock_symbol, intended_cash_amount=cash_amount)
+        sell.stock_sold_amount = cash_amount//stock_price
+        sell.actual_cash_amount = sell.stock_sold_amount*price
+        sell.timestamp = time.time()
+        user_stock = UserStock.objects.get(
+            user_id=self.user_id, stock_symbol=symbol)
+        user_stock.update_amount(sell.stock_sold_amount*-1)
+        return sell
+
+    def commit(self, user):
+        user.update_balance(self.actual_cash_amount)
+        self.save()
+
+    def cancel(self, user):
+        print("not implemented yet")
+        #TODO
+
 
 class Buy(models.Model):
     user_id = models.ForeignKey(User, on_delete=models.CASCADE)
     stock_symbol = models.CharField(max_length=3, primary_key=True)
-    cash_amount = models.DecimalField(
+    intended_cash_amount = models.DecimalField(
         max_digits=65, decimal_places=2, default=0)
+    actual_cash_amount = models.DecimalField(
+        max_digits=65, decimal_places=2, default=0)
+    stock_bought_amount = models.PositiveIntegerField(default=0)
+
+    @classmethod
+    def create(cls, user_id, stock_symbol, cash_amount, stock_price, user):
+        buy = cls(user_id=user_id, stock_symbol=stock_symbol, cash_amount=cash_amount)
+        buy.stock_bought_amount = cash_amount//stock_price
+        buy.actual_cash_amount = buy.stock_bought*price
```

Comment:
Where are buy.stock_bought and price? Is that supposed to be stock_bought_amount and stock_price?

Reply:
Yep, another good catch lol, this is why we need to be able to test!